scrapers/github.js

```diff
@@ -8,98 +11,125 @@ const REQUEST_HEADER = {
-    dataItem.followers = followersData.length;
-    dataItem.followers_data = followersData;
+    constructor() {
+        super('Github', 'https://github.com');
+    }
 
-    // user following
-    var userFollowingUrl = searchUserData.following_url.split("{")[0];
-    var followingData = getExtraUserInfo(userFollowingUrl);
-    dataItem.following = followingData.length;
-    dataItem.following_data = followingData;
+    argumentSanityCheck(args) {
+        super.argumentSanityCheck(args);
+        return true;
+    }
 
-    // starred
-    var starredUrl = searchUserData.starred_url.split("{")[0];
-    var starredData = getExtraUserInfo(starredUrl);
-    dataItem.starred = starredData.length;
-    dataItem.starred_data = starredData;
+    onInit() {
 
-    // gists
-    var gistsUrl = searchUserData.gists_url.split("{")[0];
-    dataItem.gists = getExtraUserInfo(gistsUrl);
+    }
 
-    // subscriptions
-    var subscriptionsData = getExtraUserInfo(searchUserData.subscriptions_url);
-    dataItem.subscriptions = subscriptionsData;
+    getGithubProfilesPromise(query) {
+        let requestUrl = `${GITHUB_USERS_URL}?q=${query}`;
+        let options = {
+            uri: requestUrl,
+            headers: {'User-Agent': USER_AGENT}
+        };
+        console.log(options);
+        return request(options);
+    }
 
-    // events
-    var eventsData = getExtraUserInfo(searchUserData.events_url.split("{")[0]);
-    dataItem.events = eventsData;
+    getExtraUserInfoPromise(requestUrl) {
+        let options = {
+            uri: requestUrl,
+            headers: {'User-Agent': USER_AGENT}
+        };
 
-    // received-events
-    var receivedEventsData = getExtraUserInfo(
-        searchUserData.received_events_url);
-    dataItem.received_events = receivedEventsData;
+        console.log(options);
+        return request(options);
+    }
 
-    // organizations
-    dataItem.organizations = getExtraUserInfo(
-        searchUserData.organizations_url);
+    scrape($) {
+    
+    }
 
-    // joining_date, bio, full_name, gravatar_id, home_location, works_for,
-    // email, special_link (blog)
-    var userData = getExtraUserInfo(searchUserData.url);
-    dataItem.joining_date = userData.created_at.split("T")[0];
-    dataItem.bio = userData.bio;
-    dataItem.full_name = userData.name;
-    dataItem.gravatar_id = userData.gravatar_id;
-    dataItem.home_location = userData.location;
-    dataItem.works_for = userData.company;
-    if (userData.email) {
-        dataItem.email = userData.email;
-    } else {
-        dataItem.email = "";
+    getScrapedData(query, callback) {
+        let profileData = {};
+        let matchedProfiles = Rx.Observable.fromPromise(this.getGithubProfilesPromise(query));
+        let githubProfile = {};
+        
+        matchedProfiles
+            // .do(t => console.log(t))
+            .flatMap((t,i) => {
+                githubProfile = t["items"][0]; // facing problems here
```

Comment:
Facing `TypeError: Cannot read property '0' of undefined`.
Sample API response: https://pastebin.com/0Kvd2xwK